pkg/cmd/server/kubernetes/node/node.go

```diff
@@ -82,73 +82,67 @@ func (c *NodeConfig) EnsureDocker(docker *dockerutil.Helper) {
+
+	if url, err := url.Parse(endpoint); err == nil && url.Scheme == "unix" && len(url.Path) > 0 {
 		s, err := os.Stat(url.Path)
 		switch {
 		case os.IsNotExist(err):
-			c.HandleDockerError(fmt.Sprintf("No Docker socket found at %s. Have you started the Docker daemon?", url.Path))
+			glog.Fatalf("No Docker socket found at %s. Have you started the Docker daemon?", url.Path)
 			return
 		case os.IsPermission(err):
-			c.HandleDockerError(fmt.Sprintf("You do not have permission to connect to the Docker daemon (via %s). This process requires running as the root user.", url.Path))
+			glog.Fatalf("You do not have permission to connect to the Docker daemon (via %s). This process requires running as the root user.", url.Path)
 			return
 		case err == nil && s.IsDir():
-			c.HandleDockerError(fmt.Sprintf("The Docker socket at %s is a directory instead of a unix socket - check that you have configured your connection to the Docker daemon properly.", url.Path))
+			glog.Fatalf("The Docker socket at %s is a directory instead of a unix socket - check that you have configured your connection to the Docker daemon properly.", url.Path)
 			return
 		}
 	}
 	if err := dockerClient.Ping(); err != nil {
-		c.HandleDockerError(fmt.Sprintf("Docker could not be reached at %s.  Docker must be installed and running to start containers.\n%v", dockerAddr, err))
+		glog.Fatalf("Docker could not be reached at %s.  Docker must be installed and running to start containers.\n%v", endpoint, err)
 		return
 	}
 
-	glog.Infof("Connecting to Docker at %s", dockerAddr)
+	glog.Infof("Connecting to Docker at %s", endpoint)
 
 	version, err := dockerClient.Version()
 	if err != nil {
-		c.HandleDockerError(fmt.Sprintf("Unable to check for Docker server version.\n%v", err))
+		glog.Fatalf("Unable to check for Docker server version.\n%v", err)
 		return
 	}
 
 	serverVersion, err := dockerclient.NewAPIVersion(version.APIVersion)
 	if err != nil {
-		c.HandleDockerError(fmt.Sprintf("Unable to determine Docker server version from %q.\n%v", version.APIVersion, err))
+		glog.Fatalf("Unable to determine Docker server version from %q.\n%v", version.APIVersion, err)
 		return
 	}
 
 	minimumPullByIDVersion, err := dockerclient.NewAPIVersion(minimumDockerAPIVersionWithPullByID)
 	if err != nil {
-		c.HandleDockerError(fmt.Sprintf("Unable to check for Docker server version.\n%v", err))
+		glog.Fatalf("Unable to check for Docker server version.\n%v", err)
 		return
 	}
 
 	if serverVersion.LessThan(minimumPullByIDVersion) {
-		c.HandleDockerError(fmt.Sprintf("Docker 1.6 or later (server API version %s or later) required.", minimumDockerAPIVersionWithPullByID))
+		glog.Fatalf("Docker 1.6 or later (server API version %s or later) required.", minimumDockerAPIVersionWithPullByID)
 		return
 	}
 
-	c.DockerClient = dockerClient
-}
-
-// HandleDockerError handles an an error from the docker daemon
-func (c *NodeConfig) HandleDockerError(message string) {
-	if !c.AllowDisabledDocker {
```

Comment:
why remove this?

Reply:
> why remove this?

kubelet removed the ability to pass the client through, so the responsibility for wiring a fake falls to kubelet.  We can't wire it through any more.